server/src/main/java/org/apache/iotdb/db/engine/merge/selector/MaxFileMergeFileSelector.java

```diff
@@ -213,18 +213,18 @@ private boolean checkClosed(TsFileResource unseqFile) {
   private void selectOverlappedSeqFiles(TsFileResource unseqFile) {
 
     int tmpSelectedNum = 0;
-    for (Entry<String, Long> deviceStartTimeEntry : unseqFile.getStartTimeMap().entrySet()) {
+    for (Entry<String, Integer> deviceStartTimeEntry : unseqFile.getDeviceToIndexMap().entrySet()) {
       String deviceId = deviceStartTimeEntry.getKey();
-      Long unseqStartTime = deviceStartTimeEntry.getValue();
-      Long unseqEndTime = unseqFile.getEndTimeMap().get(deviceId);
+      long unseqStartTime = unseqFile.getStartTime(deviceId);
+      long unseqEndTime = unseqFile.getEndTime(deviceId);
```

Comment:
If we already have the index info, that is `entry.getValue()`. It will be more efficient to get from endTimes by index directly. Otherwise, it will hash twice and find in bucket in `ConcurrentHashMap.` Other iterations have similar problems, please have a look too.

Reply:
Fixed